### ISSUE TITLE
Issues with IdentityUser Extra Properties UI (related entity)

### ISSUE THREAD

**Issue by gdunit** [abpframework/abp#10797]
ABP Framework: 5.0 RC1
UI: Blazor
DB: EF Core

[First, congratulations on a great job on this framework!]

Use case: I am trying to associate a related company entity to a user. The company is mandatory for each user.
However, I am finding that:
1) The company input cannot be made mandatory (despite the relevant configuration per the documents)
2) The correct company is not displayed in the drop-down when the user is opened for editing.

Database configuration:
```
 ObjectExtensionManager.Instance
    .MapEfCoreProperty<IdentityUser, Guid?>("CompanyId");
```

Entity configuration (ConfigureExtraProperties - domain.shared):
```            
     ObjectExtensionManager.Instance.Modules()
           .ConfigureIdentity(identity =>
           {
               identity.ConfigureUser(user =>
               {
                   user.AddOrUpdateProperty<Guid?>(
                       "CompanyId", //property name
                       property =>
                       {
                           property.Attributes.Add(new RequiredAttribute());
                           property.UI.Lookup.Url = "/api/app/company/name-only";
                           property.UI.Lookup.FilterParamName = "filter";
                           property.UI.Lookup.DisplayPropertyName = "name";
                           property.DisplayName = LocalizableString.Create<MyAppResource>(
                            "Admin:Users:Company"
                           );
                       }
                   );
               });
           });
```

DTO configuration:
```
      ObjectExtensionManager.Instance
        .AddOrUpdateProperty<Guid?>(
            new[]
            {
            typeof(IdentityUserDto),
            typeof(IdentityUserCreateDto),
            typeof(IdentityUserUpdateDto)
            },
            "CompanyId",
            property =>
            {
                property.Attributes.Add(new RequiredAttribute());
            });
```
The dto which is used to search companies:
```
public class CompanySearchDto : EntityDto<Guid>
    {
        public string Name { get; set; }
    }
```

Outcome: 
OK - I can select a company just fine against the user in the UI. 
OK - The drop-down is populated as I type, and when I select a company and save, it is stored OK in the database.
OK - When the table list is reloaded, the company name is displayed against the user.
KO - When I open the edit modal, the selected company name is not displayed.
KO - Additionally, if I have edited another user previously and selected their company, the selected company from that previous user is still shown when I edit a second user (even if they are assigned to a different company).
KO - If I leave the drop-down blank, then I am able to save the user anyway (despite the RequiredAttribute above in the DTO config).

My expected behaviour, unless I have misunderstood the functionality: 
- The drop-down should be populated with the name of the currently selected company when editing a user (or cleared if none)
- When the required attribute is set, the company field should be mandatory and require input prior to save

**Comment by berkansasmaz** [comment]
First of all, thank you for your good explanation.
I will test this case.

**Comment by fileman** [comment]
I have similar issue when I try to add RequiredAttribute() to IdentityUser*Dto props Name and Surmane

**Comment by berkansasmaz** [comment]
I tested this case. Here are my observations:

Command: `abp new TestGH -t app -u blazor --database-provider -csf`
Version : `v5.0.0`

> When I open the edit modal, the selected company name is not displayed.

- I could not reproduce.

> Additionally, if I have edited another user previously and selected their company, the selected company from that previous user is still shown when I edit a second user (even if they are assigned to a different company).

- I could not reproduce.

> If I leave the drop-down blank, then I am able to save the user anyway (despite the RequiredAttribute above in the DTO config).

- I could not reproduce.

![Kapture 2021-12-24 at 11 07 53](https://user-images.githubusercontent.com/31216880/147333518-047d4acc-249d-4ce8-8a7f-0631679d69bf.gif)

---

Probably these are solved in the stable version. But there is another problem for Blazor.

**I see more than one of the same items as in the picture below. I encounter this situation especially when I click on dropdown again after sending an incorrect request.**

<img width="1679" alt="Screen Shot 2021-12-24 at 11 19 51" src="https://user-images.githubusercontent.com/31216880/147334321-e5ebf8b1-c170-4d0e-bfb7-8eb46714c55d.png">

**PS:** Normally there are two items in total in the list, but more than one `Production` item appears in the dropdown.

Also, validation for client-side can be added. I opened a new issue about this => #11060

**Minimal, Reproducible Example:**
[TestGH.zip](https://github.com/abpframework/abp/files/7773494/TestGH.zip)
